src/util/psbt/mod.rs

```diff
@@ -88,6 +91,82 @@ impl PartiallySignedTransaction {
 
         Ok(())
     }
+
+    /// Decide whether the current Input spends segwit outputs or not and
+    /// calculate the corresponding Sighash.
+    /// #Panics:
+    /// Panics if the index >= number of inputs in psbt
+    pub fn signature_hash(&self, idx: usize) -> Result<SigHash, self::Error> {
+        let inp = &self.inputs[idx];
+        let sighash_type = inp.sighash_type.unwrap_or_else(|| SigHashType::All);
+        // Compute Script code for the Script. When this script is used as scriptPubkey, 
+        // the Script Code determines what goes into sighash.
+        // 1) For non-p2sh non-segwit outputs, script code is the public key.
```

Comment:
I assume the line should read "for non-segwit and non-P2SH wrapped segwit outputs" b/c in the current reading it seems that all P2SH (non-wrapped) should fall into this first options as well